src/Services/FHIR/FhirAllergyIntoleranceService.php

```diff
@@ -0,0 +1,178 @@
+                        outcome,
+                        reaction,
+                        severity_al
+                        FROM lists
+                        WHERE type = 'allergy'";
+
+        $allergyIntolerenceresults = sqlStatement($SQL);
+        $results = array();
+        while ($row = sqlFetchArray($allergyIntolerenceresults)) {
+            $codeSQL = "SELECT dx_code 
+                            FROM icd10_dx_order_code
+                            WHERE short_desc = ?;";
+            $code = sqlQuery($codeSQL, array($row['reaction']));
+            $row['code'] = $code['dx_code'];
+            array_push($results, $row);
+        }
+        return $results;
+    }
+
+    public function getOne($id)
+    {
+        $SQL = "SELECT id,
+                        date as recorded_date,
+                        type,
+                        subtype,
+                        title,
+                        begdate,
+                        enddate,
+                        returndate,
+                        referredby,
+                        extrainfo,
+                        diagnosis,
+                        pid,
+                        outcome,
+                        reaction,
+                        severity_al
+                        FROM lists
+                        WHERE type = 'allergy'
+                        AND id = ?;";
+
+        $sqlResult = sqlStatement($SQL, $id);
+        $result = sqlFetchArray($sqlResult);
+        $codeSQL = "SELECT dx_code
+                        FROM icd10_dx_order_code
+                        WHERE short_desc = ?;";
+
+        $code = sqlQuery($codeSQL, array($result['reaction']));
+        $result['code'] = $code['dx_code'];
+        return $result;        
+    }
+
+    public function createAllergyIntoleranceResource($id = '', $data = '', $encode = true)
+    {     
+        $type = new FHIRAllergyIntoleranceType();
+        $type->setValue($data['type']);
+
+        $onSetDateTime = new FHIRDateTime();
+        $onSetDateTime->setValue($data['begdate']);
+
+        $recordedDate = new FHIRDateTime();
+        $recordedDate->setValue($data['recorded_date']);
+        
+        $clinicalStatus = '';
+        if ($data['outcome'] == '1' && isset($data['enddate']))
+            $clinicalStatus = "resolved";
+        else if (isset($data['enddate'])) 
+            $clinicalStatus = "active";
+        else
+            $clinicalStatus = "inactive";
+        $clinicalStatusCoding = new FHIRCoding();
+        $clinicalStatusCoding->setSystem("http://terminology.hl7.org/CodeSystem/allergyintolerance-clinical");
+        $clinicalStatusCoding->setCode($clinicalStatus);
+        $clinicalStatusCoding->setDisplay($clinicalStatus);
+
+        $category = new FHIRAllergyIntoleranceCategory();
+        $category->setValue($data['subtype']);
+
+        $criticality = new FHIRAllergyIntoleranceCriticality();
+        $criticality->setValue($data['severity_al']);
```

Comment:
Need to set the defined codes and coding ref. You will need to translate how we record to fhir coding ie 
severity: mild to criticality: low risk etc https://www.hl7.org/fhir/codesystem-allergy-intolerance-criticality.html

and category I believe we use medication because I don't think we record anywhere.

Reply:
If we don't have data for 'category', since the cardinality is 0..1 the best practice is to omit the element (rather than arbitrarily assign a 'category' value).

Reply:
Thanks for the tip @rhausam